tap_square/client.py

```diff
@@ -295,3 +302,46 @@ def get_payments(self, start_time, bookmarked_cursor):
                 raise Exception(result.errors)
 
             yield (result.body.get('payments', []), result.body.get('cursor'))
+
+
+    def get_batch_token(self, link): #pylint: disable=no-self-use
+        if link:
+            url = link[link.find('<')+1:link.find('>')]
+            parsed = urlparse.urlparse(url)
+            batch_token = parse_qs(parsed.query)['batch_token'][0]
+            return int(batch_token)
+        return None
```

Comment:
I'm confused why this returns none and why the while loop works the way it does. What's the flow here?

Reply:
Is it supposed to be the way it paginates?

Reply:
if there is no 'Link' in the header it means that the records are finished being returned, so i want batch_token to be None, because then it will break out of the while loop.

Reply:
This actually makes sense now that i think about it. If we were using type hints this method would return an `Optional[int]` (Optional in python just means it's either an int or a None).